### DISCUSSION TITLE
How to set focus on child of ItemsRepeater item in Avalonia?

### DISCUSSION THREAD

**Discussion by NikhilSP** [AvaloniaUI/Avalonia#11511] [General]
I have itemsRepeater inside a grid (Grid has other content too)

```
<Grid RowDefinitions="*" >     
     <ItemsRepeater Grid.Row="0" Items="{Binding MyItems}" >
            <ItemsRepeater.ItemTemplate>
                <DataTemplate DataType="MyItemsUI">
                    <ContentControl  Content="{Binding}" />
                </DataTemplate>
            </ItemsRepeater.ItemTemplate>
        </ItemsRepeater>
 </Grid>
```

MyItemsUI has multiple elements inside (removed unneccessary xaml properties)

```
 <Grid>    
        <TextBox TabIndex="1" Text="{Binding FirstName}"/>   
        <TextBox TabIndex="2" Text="{Binding LastName}" />      
</Grid>
```

When there is only 1 item in ItemsRepeater -> On first Tab, focus goes on TextBox1 and on next Tab the focus goes out of the UI and not on 2nd text box

When there is multiple item in ItemsRepeater -> On first Tab, focus goes on first text box of first item and then goes out of focus. It does not go to second textbox or 2nd item's first text box.

I have tried, tabindex,tabstop=true, focusable=true, KeyBoardNavigation.TabNavigation = Continue. Nothing worked.

Any advice, what I need to do?

**Comment by timunie** [comment]
Which Avalonia version?

**Reply by NikhilSP** [reply]
10.18

**Reply by NikhilSP** [reply]
Any success? @timunie
